test/integration/user_creation_test.rb

```diff
@@ -135,10 +135,10 @@ def test_user_create_redirect
     assert_equal register_email.to[0], new_email
     # Check that the confirm account url is correct
     confirm_regex = Regexp.new("/user/redirect_tester/confirm\\?confirm_string=([a-zA-Z0-9]*)")
-    register_email.parts.each do |part|
+    email_text_parts(register_email).each do |part|
       assert_match confirm_regex, part.body.to_s
     end
-    confirm_string = register_email.parts[0].body.match(confirm_regex)[1]
+    confirm_string = email_text_parts(register_email)[0].body.match(confirm_regex)[1]
```

Comment:
`email_text_parts` returns those parts that have a content-type matching `text/*`, i.e. it includes both text and HTML parts. Really it's just there to avoid trying to match a regex against image data.

Perhaps the name is confusing. `text_and_html_parts` ??